shop/shop-service/src/main/java/shop/woowasap/shop/service/CartService.java

```diff
@@ -0,0 +1,72 @@
+package shop.woowasap.shop.service;
+
+import java.text.MessageFormat;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import shop.woowasap.core.id.api.IdGenerator;
+import shop.woowasap.shop.domain.api.cart.CartUseCase;
+import shop.woowasap.shop.domain.api.cart.request.AddCartProductRequest;
+import shop.woowasap.shop.domain.api.cart.response.CartResponse;
+import shop.woowasap.shop.domain.api.product.request.UpdateProductRequest;
+import shop.woowasap.shop.domain.cart.Cart;
+import shop.woowasap.shop.domain.cart.CartProduct;
+import shop.woowasap.shop.domain.cart.CartProductQuantity;
+import shop.woowasap.shop.domain.exception.CannotFindProductException;
+import shop.woowasap.shop.domain.product.Product;
+import shop.woowasap.shop.domain.spi.CartRepository;
+import shop.woowasap.shop.domain.spi.ProductRepository;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class CartService implements CartUseCase {
+
+    private final CartRepository cartRepository;
+    private final ProductRepository productRepository;
+    private final IdGenerator idGenerator;
+
+    @Override
+    @Transactional
+    public void updateCartProduct(final long userId,
+        final UpdateProductRequest updateProductRequest) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    @Transactional
+    public void addCartProduct(final long userId,
+        final AddCartProductRequest addCartProductRequest) {
+        if (cartRepository.existCartByUserId(userId)) {
+            cartRepository.createEmptyCart(idGenerator.generate(), userId);
+        }
```

Comment:
요거 아래 처럼 바뀌어야 하는거 아닌가요??

```suggestion
        if (!cartRepository.existCartByUserId(userId)) {
            cartRepository.createEmptyCart(idGenerator.generate(), userId);
        }
```

Reply:
맞네요! 수정하겠습니당~